### PR TITLE
DAOS-2754 control: Client should not require config file

### PR DESCRIPTION
**Pull request by mjmac**
If a file is available, it will be read. Default config
values will be used otherwise. This reduces a bit of
administrative friction in the case of running the tool
on a server, as well as in simple single-host test
scenarios.

Also adds tests for client configuration and improves
package coverage by 12%.

### CODE REVIEW

File: src/control/client/config_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+func TestLoadConfigFromDefaultFile(t *testing.T) {
+	defaultConfig := getDefaultConfig(t)
+	if err := os.MkdirAll(path.Dir(defaultConfig.Path), 0755); err != nil {
+		t.Fatal(err)
+	}
+	f, err := os.Create(defaultConfig.Path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(defaultConfig.Path)
```
Comment (tanabarr) [inline src/control/client/config_test.go]: why not just os.Remove?

File: src/control/client/config_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+func TestLoadConfigDefaultsNoFile(t *testing.T) {
+	defaultConfig := getDefaultConfig(t)
+
+	cfg, err := client.GetConfig("")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(cfg, defaultConfig) {
```
Comment (tanabarr) [inline src/control/client/config_test.go]: could use common.AssertEqual() here if you wanted.

File: src/control/client/config_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		t.Fatal(err)
+	}
+	testFile.Close()
+
+	cfg, err := client.GetConfig(testFile.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if cfg.SystemName != t.Name() {
```
Comment (tanabarr) [inline src/control/client/config_test.go]: again could use common.Assert* helpers here if you wanted

File: src/control/client/config_test.go (hunk 1 above)
Comment (mjmac) [reply src/control/client/config_test.go]: Hmm, good catch. I was thinking that it'd be the converse of MkdirAll, but of course for that to be true it should be working on a path, not the actual test file. Thinking about this some more, it's probably safest to just remove this line and trust that the whole test directory gets cleaned up.

**Pull request merged**